core/src/plugins/filed/python/postgres/BareosFdPluginPostgres.py

```diff
@@ -456,38 +494,86 @@ def close_db_connection(self):
         """
         Make sure the DB connection is closed properly. Will not throw if the connection was already closed.
         """
+
+        if self.dbCon is None:
+            return
+
         try:
             self.dbCon.close()
         except pg8000.exceptions.InterfaceError as e:
             pass
 
+    def writeAndAppendLabelFile(self, content):
+        mode = 'w' if isinstance(content, str) else 'wb'
+        with open(self.labelFileName, mode) as labelFile:
+            labelFile.write(content)
+            bareosfd.DebugMessage(150, "Did write backup_label file %s with conten:\n%s\n" % (self.labelFileName, content))
+            bareosfd.DebugMessage(
+                150, "Adding label file %s to fileset\n" % self.labelFileName
+            )
+            self.files_to_backup.append(self.labelFileName)
+            bareosfd.DebugMessage(150, "Filelist: %s\n" % (self.files_to_backup))
+
+    def writeAndAppendTablespaceMapFile(self, content):
+        mode = 'w' if isinstance(content, str) else 'wb'
+        self.tablespaceMapFileName = self.options["postgresDataDir"] + "tablespace_map"
+        with open(self.tablespaceMapFileName, mode) as tablespaceMapFile:
+            tablespaceMapFile.write(content)
+            bareosfd.DebugMessage(150, "Did write tablespace_map file %s with conten:\n%s\n" % (self.tablespaceMapFileName, content))
+            bareosfd.DebugMessage(
+                150, "Adding tablespace map file %s to fileset\n" % self.tablespaceMapFileName
+            )
+            self.files_to_backup.append(self.tablespaceMapFileName)
+            bareosfd.DebugMessage(150, "Filelist: %s\n" % (self.files_to_backup))
+
     def complete_backup_job_and_close_db(self):
         """
         Call pg_stop_backup() on PostgreSQL DB to mark the backup job as completed.
         """
         # TODO Error Handling
         # Get Backup Start Date
-        self.parseBackupLabelFile()
         # self.execute_SQL("SELECT pg_backup_start_time()")
         # self.backupStartTime = self.dbCursor.fetchone()[0]
         # Tell Postgres we are done
+
+        pgMajorVersion = self.pgVersion // 10000
+        if pgMajorVersion >= 15:
+            stopStmt = "pg_backup_stop(wait_for_archive => true)"
+        else:
+            stopStmt = "pg_stop_backup()"
+            self.parseBackupLabelFile()
+
         try:
-            results = self.dbCon.run("SELECT pg_stop_backup();")
-            self.lastLSN = self.formatLSN(results[0][0])
+            results = self.dbCon.run("SELECT %s;" % stopStmt)
+            firstRow = results[0][0]
```

Comment:
shouldn't first row being results[0] ? 
results[0][0] being the first column of the first row.